src/dmd/expressionsem.d

```diff
@@ -8709,7 +8709,32 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
             Type tn = ale.e1.type.toBasetype().nextOf();
             checkDefCtor(ale.loc, tn);
-            semanticTypeInfo(sc, tn);
+
+            Identifier hook = global.params.tracegc ? Id._d_arraysetlengthTTrace : Id._d_arraysetlengthT;
+            if (!verifyHookExist(exp.loc, *sc, hook, "resizing arrays", Id.object))
+                return setError();
+
+            // Lower to object._d_arraysetlengthT{,Trace}(e1, e2)
+            Expression id = new IdentifierExp(ale.loc, Id.empty);
+            id = new DotIdExp(ale.loc, id, Id.object);
+            id = new DotIdExp(ale.loc, id, hook);
+            id = id.expressionSemantic(sc);
```

Comment:
BTW, maybe the code above can somehow be merged into `verifyHookExists?`
```d
Expression id = verifyHookExist(exp.loc, *sc, hook, "resizing arrays", Id.object);
if (!id)
    return setError();

Reply:
Maybe rename it though and provide a way to get the hook without error reporting:

```
Expression id = findRuntimeHook(... bool printError);
```

Reply:
I don't think this PR is the correct place to change the the implementation of `verifyHookExist`.
It is already used for the already implemented hook lowerings, so I think it would be better to change it in another PR.